pkg/resource/deploy/plan_executor.go

```diff
@@ -290,6 +290,16 @@ func (pe *planExecutor) refresh(callerCtx context.Context, opts Options, preview
 		return nil
 	}
 
+	if len(opts.RefreshTargets) > 0 {
+		for _, target := range opts.RefreshTargets {
+			res := pe.plan.prev.TryGetResource(target)
```

Comment:
I think we can just use `pe.plan.olds` here:

```go
if _, ok := pe.plan.olds[target]; !ok {
```
```

Reply:
k!